lib/manager/terraform/lockfile/hash.ts

```diff
@@ -3,168 +3,129 @@ import extract from 'extract-zip';
-    );
-  } catch (err) {
-    logger.debug(
-      { err, backendLookUpName, version },
-      `Failed to retrieve builds for ${backendLookUpName} ${version}`
-    );
-    return null;
+
+  static async hashOfZipContent(
+    zipFilePath: string,
+    extractPath: string
+  ): Promise<string> {
+    await extract(zipFilePath, { dir: extractPath });
+    const files = await fs.readdir(extractPath);
+    // the h1 hashing algorithms requires that the files are sorted by filename
+    const sortedFiles = files.sort((a, b) => a.localeCompare(b));
+    const filesWithPath = sortedFiles.map((file) => `${extractPath}/${file}`);
+
+    const result = await TerraformProviderHash.hashFiles(filesWithPath);
+
+    // delete extracted files
+    await fs.rm(extractPath, { recursive: true });
+
+    return result;
   }
 
-  const builds = versionReleaseBackend.builds;
-  const hashes = await calculateHashes(builds);
+  static async calculateHashes(builds: TerraformBuild[]): Promise<string[]> {
+    const cacheDir = await ensureCacheDir('./others/terraform');
 
-  // if a hash could not be produced skip caching and return null
-  if (hashes.some((value) => value == null)) {
-    return null;
+    // for each build download ZIP, extract content and generate hash for all containing files
+    return pMap(
+      builds,
+      async (build) => {
+        const downloadFileName = join(cacheDir, build.filename);
+        const extractPath = join(cacheDir, 'extract', build.filename);
+        logger.trace(
+          `Downloading archive and generating hash for ${build.name}-${build.version}...`
+        );
+        const readStream = TerraformProviderHash.http.stream(build.url);
+        const writeStream = fs.createWriteStream(downloadFileName);
+
+        let hash = null;
+        try {
+          await fs.pipeline(readStream, writeStream);
+
+          hash = await this.hashOfZipContent(downloadFileName, extractPath);
+          logger.trace(
+            { hash },
+            `Generated hash for ${build.name}-${build.version}`
+          );
+        } catch (err) {
+          /* istanbul ignore next */
+          logger.error({ err, build }, 'write stream error');
+        } finally {
+          // delete zip file
+          await fs.unlink(downloadFileName);
+        }
+        return hash;
+      },
+      { concurrency: 4 } // allow to look up 4 builds for this version in parallel
+    );
   }
 
-  // sorting the hash alphabetically as terraform does this as well
-  const sortedHashes = hashes.sort().map((hash) => `h1:${hash}`);
-  // save to cache
-  await packageCache.set(
-    'terraform-provider-release',
-    cacheKey,
-    sortedHashes,
-    hashCacheTTL
-  );
-  return sortedHashes;
+  @cache({
+    namespace: `datasource-${TerraformProviderDatasource.id}-build-hashes`,
+    key: (registryURL: string, repository: string, version: string) =>
+      `${registryURL}/${repository}/${version}`,
+    ttlMinutes: TerraformProviderHash.hashCacheTTL,
+  })
```

Comment:
```suggestion
  @cache({
    namespace: `datasource-${TerraformProviderDatasource.id}-build-hashes`,
    key: (registryURL: string, repository: string, version: string) =>
      `${registryURL}/${repository}/${version}`,
  })
```

So only use short time cache here